Continuing top-down revisions of fuzzy storage worker doc (#553)
rspamd/rspamd.com: doc/fuzzy_storage.md

It would be prudent to compare the volume of hashes learned over some time with available RAM. For example, 400 thousands hashes may occupy about 100 Mb and 1.5 million hashes may occupy 0.5 Gb. It is not recommended to increase storage size more than the available RAM size due to a significant performance degradation. That is, don't rely on swap space, and don't choke other processes for resources. If you have a small volume of hashes suitable for learning, start with an expiration time of 90 days. Tune that down if the volume of data over that time period results in an unsuitable amount of available RAM - for example, if peak-time available RAM goes down to 20%, reduce the expiration time to 70 days, and see if data expiring from storage releases a more acceptable amount of RAM.

### Access control

> Rspamd does not allow changes to fuzzy storage by default. Any server connecting via UDP to the fuzzy_storage process must be authorized. A list of trusted IP-addresses and/or networks must be provided to make learning possible. Practically, it is better to write from the local address only (127.0.0.1) since fuzzy storage uses UDP that is not protected from source IP forgery.

Reviewer: From the perspective of the `fuzzy_storage` they are clients, not servers.

Author: Good catch, fixed.

~~~ucl
worker "fuzzy" {
  # Same options as before ...
  allow_update = ["127.0.0.1"];

  # or 10.0.0.0/8, for internal network
}
~~~

The `allow_update` setting is a comma-delimited array of strings, or a [map]({{ site.baseurl }}/doc/modules/multimap.html) of IP addresses, that are allowed to perform changes to fuzzy storage - You should also set `read_only` = no in your fuzzy_check plugin, see step 3 below.

> Transport protocol encryption might also be used for access control purposes...

Reviewer: I'd say that so far transport encryption can only distinguish between the states access/no-access but not between the states like read-only/read-write. However, it is very powerful for the first purpose and might be useful for the second one (once implemented).

Author: Line removed.

}
~~~

The  **public** `pubkey` should be copied manually to the remote host, or published in any way that guarantees the reliability (e.g. certified digital signature or HTTPS-site hosting). As always the **private** `privkey` should never be published or shared.

> Each storage can use any number of keys simultaneously, one for each remote client:

Reviewer: ```suggestion
Each storage can use any number of keys simultaneously, one for each remote client (or a group of clients):
```

}
~~~

> This mechanism is optional unless explicitly made mandatory. To enable mandatory encryption mode, add the `encrypted_only` option. Then, client systems without a valid public key are not able to access the storage in this mode.

Reviewer: ```suggestion
This mechanism is optional unless explicitly made mandatory. To enable mandatory encryption mode, add the `encrypted_only` option. In this mode, client systems that do not use a valid public key will not able to access the storage.
```
